shell/platform/darwin/ios/ios_surface.mm

```diff
@@ -39,13 +41,15 @@ bool IsIosEmbeddedViewsPreviewEnabled() {
   }
 
 #if FLUTTER_SHELL_ENABLE_METAL
-  if ([layer.get() isKindOfClass:[CAMetalLayer class]]) {
-    return std::make_unique<IOSSurfaceMetal>(
-        fml::scoped_nsobject<CAMetalLayer>(
-            reinterpret_cast<CAMetalLayer*>([layer.get() retain])),  // Metal layer
-        std::move(context),                                          // context
-        platform_views_controller                                    // platform views controller
-    );
+  if (@available(iOS kMetalOSVersionBaseline, *)) {
+    if ([layer.get() isKindOfClass:[CAMetalLayer class]]) {
+      return std::make_unique<IOSSurfaceMetal>(
+          fml::scoped_nsobject<CAMetalLayer>(
+              reinterpret_cast<CAMetalLayer*>([layer.get() retain])),  // Metal layer
+          std::move(context),                                          // context
+          platform_views_controller                                    // platform views controller
+      );
+    }
```

Comment:
Nit: `FML_CHECK(false)`  in an else clause saying this should already have been checked.

Reply:
That doesnt make sense to me, it not being available is valid since it will use the software renderer in that case